src/test/java/io/vertx/test/core/FileSystemTest.java

```diff
@@ -1712,4 +1720,144 @@ public void testAsyncFileConcurrency() throws Exception {
     }));
 
     await();
-  }}
+  }
+
+  @Test
+  public void testAtomicMove() throws Exception {
+    String source = "foo.txt";
+    String middle = "baz.txt";
+    String target = "bar.txt";
+    createFileWithJunk(source, 100);
+
+    try {
+      Files.move(new File(testDir, source).toPath(), new File(testDir, middle).toPath(), StandardCopyOption.ATOMIC_MOVE);
+    } catch (AtomicMoveNotSupportedException e) {
+      throw new AssumptionViolatedException("Atomic move unsupported");
```

Comment:
No. If the FS suports atomic moves, the test passes, otherwise it's marked as ignored.

See https://github.com/junit-team/junit4/wiki/assumptions-with-assume